more_itertools/more.py

```diff
@@ -3956,6 +3957,60 @@ def combination_index(element, iterable):
     return factorial(n + 1) // (factorial(k + 1) * factorial(n - k)) - index
 
 
+def combination_with_replacement_index(element, iterable):
+    """Equivalent to ``list(combinations_with_replacement(iterable, r)).index(element)``
+
+    The subsequences with repetition of *iterable* that are of length *r* can
+    be ordered lexicographically. :func:`combination_with_replacement_index`
+    computes the index of the first *element*, without computing the previous
+    combinations with replacement.
+
+        >>> combination_with_replacement_index('adf', 'abcdefg')
+        20
+
+    ``ValueError`` will be raised if the given *element* isn't one of the
+    combinations with replacement of *iterable*.
+    """
+    l = ilen(element)
+    element = enumerate(element)
+
+    k, y = next(element, (None, None))
+    if k is None:
+        return 0
```

Comment:
Is the idea here that if `element` is empty, then the index of the combination is 0? If that's not the right interpretation, could you add a corrected one as a comment?

Reply:
This was a copy of the function `combination_index` as seen here: https://github.com/more-itertools/more-itertools/blob/89d58c2ec089f3df54ed6cc9279949b8e2bbcd29/more_itertools/more.py#L3930-L3932
I did not put much thought as to why this is done, but I think an argument might be that both `combinations` and `combinations_with_replacement` from itertools return an empty tuple when the length of the subsequence is zero. Hence the indexing functions should return zero as that is the only element in the combinations. An example of this is then:
```python
assert list(combinations("abcd", 0)) == [()]
assert list(combinations_with_replacement("abcd", 0)) == [()]
assert list(combinations("abcd", 0))[combination_index("", "abcd")] == ()
assert list(combinations_with_replacement("abcd", 0))[combination_with_replacement_index("", "abcd")] == ()
```